File: 位运算/1734.py
from typing import List


class Solution:
    def decode(self, encoded: List[int]) -> List[int]:
        total = 0
        n = len(encoded)+1
        for i in range(1,n+1):
            total = total^i
        odd = 0
        for i in range(1,n-1,2):
            odd = odd^encoded[i]
        res = [total^odd]
        for i in range(n-1):
            res.append(res[-1]^encoded[i])
        return res

encoded = [6,5,4,6]
obj = Solution()
print(obj.decode(encoded))

File: 位运算/1734.cpp
#include<vector>
using namespace std;


class Solution {
public:
    vector<int> decode(vector<int>& encoded) {
        int total,odd;
        int n = encoded.size()+1;
        total = 0;
        for(int i=1;i<n+1;i++)
            total = total^i;
        odd = 0;
        for(int i=1;i<n-1;i+=2)
            odd = odd^encoded[i];
        vector<int>res;
        res.push_back(total^odd);
        for(int i=0;i<n-1;i++)
            res.push_back(res.back()^encoded[i]);
        return res;
    }
};